### DISCUSSION TITLE
Why cilium doesn't add routes when it runs in native mode in multi-cluster setup?

### DISCUSSION THREAD

**Discussion by Noksa** [cilium/cilium#36505] [General]
Hello.

Probably, it can be a stupid question but anyway I'd like to ask it :)

What I want to setup:
* 2 k8s clusters from different networks
* Two 'external' frr paired with each other as static neighbor (one for first k8s cluster, another one for second k8s cluster)
* All nodes from the clusters can reach each other using their IP addresses
* The goal is to have pod-to-pod connectivity

When I install cilium in native mode with bgp control plane enabled, I see that PodCIDRs are advertised to frr and they share routes between each other.
I also see that cilium in each k8s cluster contains proper bgp peers/routes but it doesn't add them to route on hosts.

Is it possible to do that just using cilium? Or I have to use kube-router or something like that?

I can get pod-to-pod connectivity working using cilium tunnel with vxlan or using native mode + kube-router but I wonder if it possible to do that in native mode without external software

**Comment by harsimran-pabla** [comment]
> I also see that cilium in each k8s cluster contains proper bgp peers/routes but it doesn't add them to route on hosts.

Cilium BGP control plane does not install learned routes in kernel. Newer versions should not even accept routes. For pod-to-pod connectivity, you'll need default route pointing towards connected routers. Upstream peers should have pod-cidr reachability since Cilium BGP advertises those routes.

**Comment by Noksa** [comment]
@harsimran-pabla makes sense. Thanks <3
